core/src/main/java/com/graphhopper/routing/AlternativeRouteCH.java

```diff
@@ -76,22 +76,21 @@ public boolean apply(final int v, final SPTEntry fromSPTEntry) {
                 // s -> v and v -> t need not be shortest paths. In fact, they can sometimes be pretty strange.
                 // We still use this preliminary path to filter for shared path length with other alternatives,
                 // so we don't have to work so much.
-                Path preliminaryRoute = createPathExtractor(graph, weighting).extract(fromSPTEntry, toSPTEntry, fromSPTEntry.getWeightOfVisitedPath() + toSPTEntry.getWeightOfVisitedPath());
+                Path preliminaryRoute = createPathExtractor(graph).extract(fromSPTEntry, toSPTEntry, fromSPTEntry.getWeightOfVisitedPath() + toSPTEntry.getWeightOfVisitedPath());
                 double preliminaryShare = calculateShare(preliminaryRoute);
                 if (preliminaryShare > maxShareFactor) {
                     return true;
                 }
 
                 // Okay, now we want the s -> v -> t shortest via-path, so we route s -> v and v -> t
                 // and glue them together.
-                DijkstraBidirectionCHNoSOD svRouter = new DijkstraBidirectionCHNoSOD(graph, weighting);
-                svRouter.setEdgeFilter(additionalEdgeFilter);
+                DijkstraBidirectionCHNoSOD svRouter = new DijkstraBidirectionCHNoSOD(graph);
                 final Path svPath = svRouter.calcPath(s, v);
                 final IntIndexedContainer svNodes = svPath.calcNodes();
-                DijkstraBidirectionCHNoSOD vtRouter = new DijkstraBidirectionCHNoSOD(graph, weighting);
-                vtRouter.setEdgeFilter(additionalEdgeFilter);
+                DijkstraBidirectionCHNoSOD vtRouter = new DijkstraBidirectionCHNoSOD(graph);
                 final Path vtPath = vtRouter.calcPath(v, t);
                 final IntIndexedContainer vtNodes = vtPath.calcNodes();
+                // todonow: which graph do we want here ?
```

Comment:
@michaz ? Do we wan the BaseGraph, the QueryGraph (of the CHGraph) or the CHGraph ?

Reply:
Well, the Path will be a path through the .. well, the graph. :-) Doesn't use shortcuts, but _does_ use the query-graph nodes.

Technically, the QueryGraph of the BaseGraph, I guess. But the QueryGraph of the CHGraph works as well, I think.

Reply:
So you need virtual nodes + edges but you do *not* want shortcuts ?

Reply:
Correct, but I think I don't mind _if there are_ shortcuts, I just don't use them.

The graph at this point is just used because the constructor of Path wants it, and the Path will not have shortcuts. Because I concatenate it from two "expanded" Paths from the CH-router.

Reply:
If (as you say) a CHGraph is ultimately something _different_ than a Graph, then this class must conceptually probably get the CHGraph _and_ the Graph, because it calls sub-routers (with the CHGraph), but it also wants to reason about the (pure) Graph.

Reply:
It looks like the overriden `.extract()` method does not use the `graph` at all ? Do you really have to use the Path this way ? Can we not just get a list of edgeiteratorstates + total distance or something ?

Reply:
Ah never mind the path is passed to another method that calls calcEdges (which uses the graph) ...

Reply:
Yeah, the overridden extract is just a relic from something that was there before. If I don't need the Graph at all here, throw it out, no problem.

But I do want a Path. A Path through a Graph is something we should have as an element, without having to chop it up into bits.

Reply:
Ok I changed to `graph.getGraph().getBaseGraph()` (sigh). This will be the query graph of the base graph, but we need to get rid of this construct. 

> A Path through a Graph is something we should have as an element, without having to chop it up into bits

? Can you explain what you mean here ? Where is the path chopped up into bits ?

Reply:
There should also be some tests for alternative route CH, maybe checking that all returned paths start and end in the right places and the weight is less than twice as big as the best one or something like this ?